File: app/src/main/java/br/com/nicetuber/base/BasePresenter.java
package br.com.nicetuber.base;

import javax.inject.Inject;

/**
 * Created by thiagozg on 14/10/2017.
 */

public class BasePresenter<V extends IView, I extends BaseInteractor> {

    protected V view;

    @Inject
    protected I interactor;

    public void attachView(V view) {
        this.view = view;
    }

    public void detachView() {
        this.view = null;
    }

}

File: app/src/main/java/br/com/nicetuber/base/BasePresenter.kt
package br.com.nicetuber.base

import javax.inject.Inject

/**
 * Created by thiagozg on 14/10/2017.
 */

open class BasePresenter<V : IView, I : BaseInteractor> {

    protected var view: V? = null

    @Inject
    protected lateinit var interactor: I

    fun attachView(view: V) {
        this.view = view
    }

    fun detachView() {
        this.view = null
    }

}
